basic-services/ReadPropertyResult.cs

```diff
@@ -39,42 +40,130 @@ public struct ReadPropertyResult
+                        NumericValue = 1;
+                        StringValue = "True";
+                        BooleanValue = true;
+                    } else {
+                        NumericValue = 0;
+                        StringValue = "False";
+                        BooleanValue = false;
+                    }
+                    break;
+                case JTokenType.Object:
+                    ReadObject(token);
+                    break;
                 default:
-                    StringValue = null;
-                    NumericValue = 0;
-                    ArrayValue = null;
-                    BooleanValue = false;
+                    NumericValue = 1;
+                    StringValue = "Unsupported Data Type";
                     break;
             }
-            
         }
 
-    }
-
-    
+        /// <summary>Parses a JArray and adds (string, double) pairs to an array based on JToken type.</summary>
+        internal void ReadArray(JToken token) {
+            JArray arr = JArray.Parse(token.ToString());
+            ArrayValue = new (string, double, bool)[arr.Count];
+            int index = 0;
+            foreach(var item in arr.Children())
+            {
+                double value = 0;
+                switch (item.Type)
+                {
+                    case JTokenType.Integer:
+                        value = (double) item.Value<double>();
+                        ArrayValue[index] = (value.ToString(), value, Convert.ToBoolean(NumericValue));
+                        break;
+                    case JTokenType.Float:
+                        value = (double) item.Value<double>();
+                        ArrayValue[index] = (value.ToString(), value, Convert.ToBoolean(NumericValue));
+                        break;
+                    case JTokenType.String:
+                        ArrayValue[index] = (item.Value<string>(), value, false);
+                        break;
+                    default:
+                        ArrayValue[index] = ("Unsupported Data Type", 1, false);
+                        break;
+                }
+                index++;
+            }
+            NumericValue = 0;
+            StringValue = "Array";
+        }
 
+        /// <summary>Searches the JObject for reliability and priority fields and finds a value to use.</summary>
+        internal void ReadObject(JToken token) {
+            JToken valueToken = token["value"];
+            JToken reliabilityToken = token["reliability"];
+            JToken priorityToken = token["priority"];
 
-}
\ No newline at end of file
+            if (reliabilityToken != null) 
+            {
+                Reliability = reliabilityToken.ToString();
+            }
+            if (priorityToken != null)
+            {
+                Priority = priorityToken.ToString();
+            }
+            if (valueToken == null) {
+                // Search for the first value to use as the value (there could exist more than 1)
```

Comment:
I don't think there can be more than one. You can just search for the property named `"value"`.